src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.AbstractTest;
+import mate.academy.dao.RoleDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoImplTest extends AbstractTest {
+    private RoleDao roleDao;
+    private Role role;
+
+    @BeforeEach
+    void setUp() {
+        role = new Role(Role.RoleName.USER);
+        roleDao = new RoleDaoImpl(getSessionFactory());
+    }
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[] {Role.class};
+    }
+
+    @Test
+    void saveRole_Ok() {
+        Role actual = roleDao.save(role);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(role, actual);
+        Assertions.assertEquals(1L, actual.getId());
+        Assertions.assertEquals(Role.RoleName.USER, actual.getRoleName());
```

Comment:
Please use static imports for methods to make code cleaner when it's obvious from which class method is called
```suggestion
    void saveRole_Ok() {
        Role actual = roleDao.save(role);
        assertNotNull(actual);
        assertEquals(role, actual);
        assertEquals(1L, actual.getId());
        assertEquals(Role.RoleName.USER, actual.getRoleName());
```